stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/TextAndAttachmentsViewHolder.kt

```diff
@@ -71,36 +72,47 @@ internal class PlainTextWithFileAttachmentsViewHolder(
     override fun bindData(data: MessageListItem.MessageItem, diff: MessageListItemPayloadDiff?) {
         super.bindData(data, diff)
 
+        binding.messageText.isVisible = data.message.text.isNotEmpty()
         markdown.setText(binding.messageText, data.message.text)
-        setupAttachments(data)
 
-        val uploadIdList: List<String> = data.message
-            .attachments
-            .filter { attachment -> attachment.uploadState == Attachment.UploadState.InProgress }
-            .mapNotNull { attachment -> attachment.uploadId }
+        setupAttachment(data)
+        setupUploads(data)
+    }
 
-        val needUpload = uploadIdList.isNotEmpty()
+    private fun setupAttachment(data: MessageListItem.MessageItem) {
+        binding.attachmentsContainer.removeAllViews()
```

Comment:
All related attachments code will go to the attachment view factory. I'm going to make `AttachmentViewFactory` easier only with one method for creating attachment content view (some `ViewGroup` that contain both link attachment and other attachment type view)